src/driver/mod.rs

```diff
@@ -92,10 +94,10 @@ pub fn nsm_process_request(fd: i32, request: Request) -> Response {
         return Response::Error(ErrorCode::InputTooLarge);
     }
 
-    let mut cbor_response: [u8; NSM_RESPONSE_MAX_SIZE] = [0; NSM_RESPONSE_MAX_SIZE];
+    let cbor_response: [u8; NSM_RESPONSE_MAX_SIZE] = [0; NSM_RESPONSE_MAX_SIZE];
     let mut message = NsmMessage {
         request: IoSlice::new(&cbor_request),
-        response: IoSliceMut::new(&mut cbor_response),
+        response: IoSlice::new(&cbor_response),
```

Comment:
This didn't appear to need to be `mut`, even in the previous implementation. I planned on relying on the test suite to prove me wrong but was unable to successfully follow the testing instructions (see PR description)